persistent/persistence.py

```diff
@@ -95,12 +98,18 @@ def _get_oid(self):
     def _set_oid(self, value):
         if value == self.__oid:
             return
-        if value is not None:
-            if not isinstance(value, OID_TYPE):
-                raise ValueError('Invalid OID type: %s' % value)
-        if self.__jar is not None and self.__oid is not None:
-            raise ValueError('Already assigned an OID by our jar')
-        self.__oid = value
+        # The C implementation allows *any* value to be
+        # used as the _p_oid.
```

Comment:
There are definitely test cases in ZODB that require setting `_p_oid` to non-byte values (typically integers, IIRC). I think our own organization's code base also does that.